src/tamm/tamm_utils.hpp

```diff
@@ -3009,13 +3004,9 @@ Tensor<TensorType> tensor_block(Tensor<TensorType> tensor, std::vector<int64_t>
   GA_Pgroup_set_default(ga_pg_default);
 #endif
 
-  Tensor<TensorType> pbtensor = btensor;
-  if(!permute.empty()) {
-    pbtensor = permute_tensor<TensorType>(btensor, permute);
-    Tensor<TensorType>::deallocate(btensor);
-  }
+  if(!permute.empty()) btensor = permute_tensor<TensorType>(btensor, permute);
```

Comment:
@jcorbino-lbl This line needs to be reverted to the previous code. `permute_tensor` returns a new tensor. Assigning it to the existing handle `btensor` results in a leak since the original tensor to which the handle `btensor` was pointing to was never deallocated.